vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -441,6 +475,19 @@ func PowerOn(vm *object.VirtualMachine) error {
 	log.Printf("[DEBUG] Powering on virtual machine %q", vm.InventoryPath)
 	ctx, cancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
 	defer cancel()
+
+	err := blockUntilReadyForMethod("PowerOnVM_Task", vm, provider.DefaultAPITimeout)
+	if err != nil {
+		return err
+	}
+
+	// This is the controversial part. Although we take every precaution to make sure the VM
+	// is in a state that can be started we have noticed that vsphere will randomly fail to
+	// power on the vm with "InvalidState" errors.
+	//
+	// We're adding a small delay here to avoid this issue.
+	time.Sleep(500 * time.Millisecond)
```

Comment:
Should we pull this out into a variable so it can be configured at the top of the file? We might need to play around with the length of time we wait until we get it right.